File: src/test/java/no/fintlabs/integration/validation/constraints/ReferencedConfigurationExistsValidatorTest.java
package no.fintlabs.integration.validation.constraints;

import no.fintlabs.integration.model.dtos.ConfigurationDto;
import no.fintlabs.integration.validation.IntegrationValidationContext;
import org.hibernate.validator.constraintvalidation.HibernateConstraintValidatorContext;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertFalse;
import static org.junit.jupiter.api.Assertions.assertTrue;
import static org.mockito.Mockito.mock;
import static org.mockito.Mockito.when;

public class ReferencedConfigurationExistsValidatorTest {

    @Test
    public void testValidator() {
        HibernateConstraintValidatorContext context = mock(HibernateConstraintValidatorContext.class);
        IntegrationValidationContext validationContext = mock(IntegrationValidationContext.class);

        ReferencedConfigurationExistsValidator validator = new ReferencedConfigurationExistsValidator();

        when(context.getConstraintValidatorPayload(IntegrationValidationContext.class)).thenReturn(validationContext);
        assertTrue(validator.isValid(null, context));

        when(context.getConstraintValidatorPayload(IntegrationValidationContext.class)).thenReturn(validationContext);
        assertFalse(validator.isValid(1L, context));

        when(validationContext.getConfiguration()).thenReturn(
                ConfigurationDto.builder()
                        .id(1L)
                        .integrationId(2L)
                        .integrationMetadataId(3L)
                        .completed(true)
                        .comment("Test Comment")
                        .version(1)
                        .build());
        assertTrue(validator.isValid(1L, context));
    }
}

File: src/test/kotlin/no/novari/flyt/integration/validation/constraints/ReferencedConfigurationExistsValidatorTest.kt
package no.novari.flyt.integration.validation.constraints

import no.novari.flyt.integration.api.dto.ConfigurationDto
import no.novari.flyt.integration.validation.IntegrationValidationContext
import org.hibernate.validator.constraintvalidation.HibernateConstraintValidatorContext
import org.junit.jupiter.api.Assertions.assertFalse
import org.junit.jupiter.api.Assertions.assertTrue
import org.junit.jupiter.api.Test
import org.mockito.kotlin.mock
import org.mockito.kotlin.whenever

class ReferencedConfigurationExistsValidatorTest {
    @Test
    fun shouldValidateReferencedConfigurationExistence() {
        val context = mock<HibernateConstraintValidatorContext>()
        val validationContext = mock<IntegrationValidationContext>()
        val validator = ReferencedConfigurationExistsValidator()

        whenever(context.getConstraintValidatorPayload(IntegrationValidationContext::class.java))
            .thenReturn(validationContext)

        assertTrue(validator.isValid(null, context))
        assertFalse(validator.isValid(1L, context))

        whenever(validationContext.configuration)
            .thenReturn(
                ConfigurationDto(
                    id = 1L,
                    integrationId = 2L,
                    integrationMetadataId = 3L,
                    completed = true,
                    comment = "Test Comment",
                    version = 1,
                ),
            )

        assertTrue(validator.isValid(1L, context))
    }
}
